workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Worker;
+using Improbable.Worker.Core;
+using Unity.Entities;
+using UnityEngine;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.Core
+{
+    public class Worker : IDisposable
+    {
+        internal readonly Dictionary<EntityId, Entity> EntityMapping = new Dictionary<EntityId, Entity>();
+
+        public readonly Vector3 Origin;
+        public readonly string WorkerType;
+        public readonly string WorkerId;
+        public readonly World World;
+        public readonly Entity WorkerEntity;
+        public readonly Connection Connection;
+        public readonly ILogDispatcher LogDispatcher;
+
+        public delegate void WorkerEventHandler(Worker worker);
+        public static event WorkerEventHandler OnConnect;
+        public static event WorkerEventHandler OnDisconnect;
+
+        private Worker(ConnectionConfig config, ILogDispatcher logDispatcher, Vector3 origin)
+        {
+            Origin = origin;
+            WorkerId = string.IsNullOrEmpty(config.WorkerId)
+                ? $"{config.WorkerType}-{Guid.NewGuid()}"
+                : config.WorkerId;
+
+            Connection = ConnectionUtility.Connect(config, WorkerId);
```

Comment:
We don't really want to run these async tasks in a constructor. We will need to replace these connection utilities with ones that actually show that this is async and let users handle it correctly.